libtenzir/builtins/operators/python.cpp

```diff
@@ -354,15 +377,32 @@ class plugin final : public virtual operator_plugin<python_operator> {
   }
 
   auto parse_operator(parser_interface& p) const -> operator_ptr override {
-    auto command = std::string{};
+    auto command = std::optional<located<std::string>>{};
     auto requirements = std::string{};
+    auto filename = std::optional<located<std::string>>{};
     auto parser = argument_parser{"python", "https://docs.tenzir.com/next/"
                                             "operators/transformations/python"};
     parser.add("-r,--requirements", requirements, "<requirements>");
+    parser.add("-f,--file", filename, "<filename>");
     parser.add(command, "<command>");
     parser.parse(p);
+    if (!filename && !command) {
+      diagnostic::error("must have either the `--file` argument or inline code")
+        .throw_();
+    }
+    if (filename && command) {
+      diagnostic::error(
+        "cannot have `--file` argument together with inline code")
```

Comment:
Here you can use `.primary(*filename).primary(*command)` to highlight both visually in the generated diagnostic. Both are primary since they both equally contribute to the diagnostic.